_books/ion-1-1/src/modules.md

```diff
@@ -1,5 +1,121 @@
+
+<!-- TODO: `export` -->
+
+### `symbol_table`
+
+The `symbol_table` clause assembles a list of text values for the module to export. It takes any number of arguments.
+
+#### Syntax
+```ion
+(symbol_table arg1 arg2 ... argN)
+```
+
+#### Processing
+
+When the `symbol_table` clause is encountered, the reader constructs an empty list. The arguments to the clause are then processed from left to right.
+
+For each `arg`:
+* **If the `arg` is a list of text values**, the nested text values are appended to the end of the symbol table being constructed.
+  * When `null`, `null.string`, `null.symbol`, or `$0` appear in the list of text values, this creates a symbol with unknown text.
+  * The presence of any other Ion value in the list raises an error.
+* **If the `arg` is the name of a module**, the symbols in that module's symbol table are appended to the end of the symbol table being constructed.
+* **If the `arg` is anything else**, the reader must raise an error.
+
+#### Example `symbol_table`
+
+```ion
+(symbol_table         // Constructs an empty symbol table (list)
+  ["a", b, 'c']       // The text values in this list are appended to the table
+  foo                 // Module `foo`'s symbol table values are appended to the table
+  ['''g''', "h", i])  // The text values in this list are appended to the table
+```
+If module `foo`'s symbol table were `[d, e, f]`, then the symbol table defined by the above clause would be:
+```ion
+["a", "b", "c", "d", "e", "f", "g", "h", "i"]
+```
+
+### `macro_table`
+
+The `macro_table` clause assembles a list of macro definitions for the module to export. It takes any number of arguments.
+
+#### Syntax
+```ion
+(macro_table arg1 arg2 ... argN)
+```
+#### Processing
+
+When the `macro_table` clause is encountered, the reader constructs an empty list. The arguments to the clause are then processed from left to right.
+
+For each `arg`:
+* **If the `arg` is a `macro` clause**, the clause is processed and the resulting macro definition is appended to the end of the macro table being constructed.
+* **If the `arg` is the name of a module**, the macro definitions in that module's macro table are appended to the end of the macro table being constructed.
+* **If the `arg` is anything else**, the reader must raise an error.
+
+
+Macro definitions being added to the macro table must have a unique name. If a macro is added whose name conflicts with one already present in the table, the reader must raise an error.
+
+### `macro`
+
+The `macro` clause defines a new macro.
+
+#### Syntax
+```ion
+(macro name signature template)
+```
+
+For more information, see _[Defining macros](macros/defining_macros.md)_.
+
+<!-- TODO: `import` -->
+<!-- TODO: `module` -->
+<!-- TODO: `export` -->
+
+## Grammar
+
+Literals appear in `code blocks`. Terminals are described in _italic text_.
+
+| Production           |     | Body                                                   |
```

Comment:
Does not necessarily need to be fixed now, but IIRC, we had something in the ascii-doc version that turned this into a neatly formatted grammar block. In markdown, this is going to appear as a table with all of the cell borders, etc.

It might be easier to read if we rewrite this as a code block rather than a table.

Reply:
I initially did that. The nice thing about the table is it allowed me to use backticks for literal tokens. In a code block, there aren't backticks, bold, italics, or other common ways of distinguishing text.

I'll think on it some more before making `{{#include ...}}`-able grammar snippets.